frontend/app/src/utils/indexed-db.ts

```diff
@@ -0,0 +1,78 @@
+export default class IndexedDb {
+  dbName: string = '';
+  dbVersion: number = 0;
+  store: string = '';
+
+  constructor(dbName: string, dbVersion: number, store: string) {
+    this.dbName = dbName;
+    this.dbVersion = dbVersion;
+    this.store = store;
+  }
+
+  get db(): Promise<IDBDatabase> | null {
+    return new Promise((resolve, reject) => {
+      if (!window.indexedDB) {
+        reject('Unsupported indexedDB');
+      }
+
+      const request = (
+        window.indexedDB ||
+        (window as any).mozIndexedDB ||
+        (window as any).webkitIndexedDB
+      ).open(this.dbName, this.dbVersion);
+
+      request.onsuccess = e => {
+        resolve((e.target as any).result);
+      };
+      request.onerror = e => {
+        reject(e);
+      };
+      request.onupgradeneeded = e => {
+        const db = (e.target as any).result;
+        if (!db.objectStoreNames.contains(this.store)) {
+          db.createObjectStore(this.store, {
+            keyPath: 'id',
+            autoIncrement: true
+          });
+        }
+      };
+
+      request.onblocked = e => {
+        reject(e);
+      };
+    });
+  }
+
+  async add(data: any, callback: Function = () => {}) {
+    if (this.db) {
+      const db = await this.db;
+      const request = db
+        .transaction(this.store, 'readwrite')
+        .objectStore(this.store)
+        .put(data);
+
+      request.onerror = (e: any) => callback(e.target.error);
+    }
+  }
+
+  async getAll(callback: Function) {
+    if (this.db) {
```

Comment:
nit: maybe could return here just to avoid nesting the code.

Also can we call the callback with a custom error if the check fails?

Reply:
I think I implemented it wrongly, I should use try-catch here instead (?)